hello.cpp

```diff
@@ -0,0 +1,46 @@
+#include <thread>
+#include <boost/asio.hpp>
+
+#include "packet.h"
+
+using boost::asio::ip::tcp;
+
+const int port = 3000;
+
+void session(tcp::socket socket, const std::shared_ptr<KeyValueStore>& k) {
+  boost::system::error_code ec;
+  try {
+    for (;;) {
+      char data[Packet::HEADER_LENGTH] = { 0 };
+      size_t bytes_read = socket.read_some(boost::asio::buffer(data), ec);
+      if (ec == boost::asio::error::eof) {
```

Comment:
This if/else chain is a little quirky, maybe combine to:
```c++
if (ec && ec != eof) 
    throw system_error(ec);
else if (ec == eof || data[0] != MAGIC || bytes_read < Packet::HEADER_LENGTH)
    return;
```
Or just remove the else statements from the return cases:

```c++
if (ec == eof)
    return;
if (ec)
   throw system_error(ec);
if (data[0] != Packet::MAGIC || bytes_read < Packet::HEADER_LENGTH)
   return;
```

I think this is an acceptable case where curly braces are not needed.